concordium-std/src/impls.rs

```diff
@@ -575,61 +576,76 @@ fn prepend_generic_map_key<K: Serial>(key: K) -> Vec<u8> {
     key_with_map_prefix
 }
 
-impl HasContractStateLL for ContractStateLL {
-    type ContractStateData = ();
-    type EntryType = StateEntry;
-    type IterType = ContractStateIter;
+impl<T> ContractStateLL<T>
+where
+    T: ContractStateApi,
+{
+    pub fn new(state_api: T) -> Self {
+        Self {
+            locked_subtrees: Default::default(),
```

Comment:
I don't think the std should keep track of locked trees. THe host already does that.